src/page-container/usePageManager.jsx

```diff
@@ -0,0 +1,200 @@
+import React from 'react';
+
+import Logger from '../utils/logger';
+import {
+  getPersistentScrollMap,
+  applyScrollData,
+} from '../utils/scroll-persistence/scroll-persistence';
+
+import PageManagerContext from './PageManagerContext';
+
+/**
+ * Hook used to manage Page registrations within a PageContainer.
+ * @param {Object} rootContainerRef A React ref object with a current value pointing
+ * to a valid HTMLElement
+ * @returns An object containing data related to currently active/visible Page
+ * and a unique PageManager component definition
+ */
+const usePageManager = (rootContainerRef) => {
+  const pageRegisterRef = React.useRef({});
+  const lastActivePageArrayRef = React.useRef([]);
+  const [activePageArray, setActivePageArray] = React.useState([]);
+
+  const pageManagerContextValue = React.useMemo(() => {
+    const registerPage = ({
+      portalElement,
+      pageId,
+      parentPageId,
+      label,
+      metaData,
+    }) => {
+      pageRegisterRef.current[pageId] = {
+        pageId,
+        portalElement,
+        parentPageId,
+        label,
+        metaData,
+        overflowData: pageRegisterRef.current[pageId]
```

Comment:
nitpick - could do this too
```suggestion
        overflowData: pageRegisterRef?.current[pageId]?.overflowData
```